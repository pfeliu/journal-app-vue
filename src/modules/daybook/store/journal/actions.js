// export const myAction = async({ commit }) => {

import journalApi from '@/api/journalApi';

// }

export const loadEntries = async ({ commit }) => {
  const { data } = await journalApi.get('/entries.json');
  // console.log(data);

  if (!data) {
    commit('setEntries', []);
    return;
  }
  const entries = [];
  for (let id of Object.keys(data)) {
    entries.push({ id, ...data[id] });
  }

  console.log(entries);
  commit('setEntries', entries);
};
export const updateEntry = async ({ commit }, entry) => {
  const { date, picture, text } = entry;
  const dataToSave = { date, picture, text };

  await journalApi.put(`/entries/${entry.id}.json`, dataToSave);

  commit('updateEntry', { ...entry });
};
export const createEntry = async ({ commit }, entry) => {
  console.log(entry);
  const { date, picture, text } = entry;
  const dataToSave = { date, picture, text };

  const { data } = await journalApi.post(`/entries.json`, dataToSave);

  dataToSave.id = data.name;

  commit('addEntry', dataToSave);

  return data.name;
};
export const deleteEntry = async ({ commit }, id) => {
  await journalApi.delete(`/entries/${id}.json`);

  commit('deleteEntry', id);

  return null;
};
