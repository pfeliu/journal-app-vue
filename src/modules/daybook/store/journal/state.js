export default () => ({
  isLoading: true,
  entries: [
    {
      id: new Date().getTime(),
      date: new Date().toDateString(),
      text: 'Aliqua duis magna aliqua reprehenderit aliquip ipsum minim deserunt eiusmod aliqua ex aliqua. Exercitation id labore laborum ex tempor sint dolor incididunt commodo fugiat deserunt consequat sunt proident. Dolor velit tempor et aliqua pariatur.',
      picture: null,
    },
    {
      id: new Date().getTime() + 1000,
      date: new Date().toDateString(),
      text: 'Aliqua duis magna aliqua reprehenderit aliquip ipsum minim deserunt eiusmod aliqua ex aliqua. Exercitation id labore laborum ex tempor sint dolor incididunt commodo fugiat deserunt consequat sunt proident. Dolor velit tempor et aliqua pariatur.',
      picture: null,
    },
    {
      id: new Date().getTime() + 2000,
      date: new Date().toDateString(),
      text: 'Aliqua duis magna aliqua reprehenderit aliquip ipsum minim deserunt eiusmod aliqua ex aliqua. Exercitation id labore laborum ex tempor sint dolor incididunt commodo fugiat deserunt consequat sunt proident. Dolor velit tempor et aliqua pariatur.',
      picture: null,
    },
  ],
});
