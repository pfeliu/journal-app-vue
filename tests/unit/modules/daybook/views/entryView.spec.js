import { createStore } from 'vuex';
import { journalState } from '../../../mock-data/test-journal-state';
import { shallowMount } from '@vue/test-utils';
import journal from '@/modules/daybook/store/journal';
import EntryView from '@/modules/daybook/views/EntryView.vue';
import Swal from 'sweetalert2';

const createVuexStore = (initialState) =>
  createStore({
    modules: {
      journal: {
        ...journal,
        state: { ...initialState },
      },
    },
  });

const store = createVuexStore(journalState);
store.dispatch = jest.fn();

const mockRouter = {
  push: jest.fn(),
};

jest.mock('sweetalert2', () => ({
  fire: jest.fn(),
  showLoading: jest.fn(),
  close: jest.fn(),
}));

describe('Entry View', () => {
  let wrapper;
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallowMount(EntryView, {
      props: {
        id: '-MfKM6PrX3s9QqURdLx5',
      },
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    });
  });

  test('should get out user when id not match', () => {
    // eslint-disable-next-line no-unused-vars
    const wrapper = shallowMount(EntryView, {
      props: {
        id: 'Id no exist',
      },
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    });

    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' });
  });

  test('should show entry correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
    expect(mockRouter.push).not.toHaveBeenCalled();
  });

  test('should delete entry and exit', async (done) => {
    Swal.fire.mockReturnValueOnce(await Promise.resolve({ isConfirmed: true }));
    wrapper.find('.btn-danger').trigger('click');

    expect(Swal.fire).toHaveBeenCalledWith({
      title: 'Are you sure?',
      text: 'It is not possible revert this action',
      showDenyButton: true,
      confirmButtonText: 'Yes, I am sure',
    });

    setTimeout(() => {
      expect(store.dispatch).toHaveBeenCalledWith(
        'journal/deleteEntry',
        '-MfKM6PrX3s9QqURdLx5'
      );
      expect(mockRouter.push).toHaveBeenCalled();
      done();
    }, 1);
  });
});
