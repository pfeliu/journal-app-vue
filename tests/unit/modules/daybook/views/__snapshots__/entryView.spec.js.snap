// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry View should show entry correctly 1`] = `
<div class="entry-title d-flex justify-content-between p-2">
  <div><span class="text-success fs-3 fw-bold">23</span><span class="mx-1 fs-3">Julio</span><span class="mx-2 fs-4 fw-light">2021, Viernes</span></div>
  <div><button class="btn btn-danger mx-2"> Delete <i class="fa fa-trash-alt"></i></button><input type="file" accept="image/png, image/jpeg" style="display: none;"><button class="btn btn-primary mx-2"> Upload photo <i class="fa fa-upload"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column px-3 h-75"><textarea>What's happening?</textarea></div>
<img src="https://res.cloudinary.com/dx0pryfzn/image/upload/v1627077252/ip3vwn8sjvnncphqka4y.jpg" alt="entry-picture" class="img-thumbnail">
<!--v-if-->
<fab-stub icon="fa-save"></fab-stub>
`;
